src/core/restrict-ifaces.c

```diff
@@ -0,0 +1,203 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#if BPF_FRAMEWORK
+#include <bpf/bpf.h>
+#include <bpf/libbpf.h>
+
+#include "bpf-link.h"
+#endif
+
+#include "fd-util.h"
+#include "restrict-ifaces.h"
+#include "socket-netlink.h"
+
+#if BPF_FRAMEWORK
+/* libbpf, clang and llc compile time dependencies are satisfied */
+
+#include "bpf/restrict_ifaces/restrict-ifaces.skel.h"
+
+static struct restrict_ifaces_bpf *restrict_ifaces_bpf_free(struct restrict_ifaces_bpf *obj) {
+        restrict_ifaces_bpf__destroy(obj);
+        return NULL;
+}
+
+DEFINE_TRIVIAL_CLEANUP_FUNC(struct restrict_ifaces_bpf *, restrict_ifaces_bpf_free);
+
+static int prepare_restrict_ifaces_bpf(Unit* u, bool is_allow_list,
+                const Set *restrict_network_interfaces,
+                struct restrict_ifaces_bpf **ret_object) {
+        _cleanup_(restrict_ifaces_bpf_freep) struct restrict_ifaces_bpf *obj = NULL;
+        char *iface;
+        uint8_t dummy = 0;
+        int ifindex;
+        int r, map_fd;
+
+        assert(ret_object);
+
+        obj = restrict_ifaces_bpf__open();
+        if (!obj)
+                return log_unit_error_errno(u, SYNTHETIC_ERRNO(ENOMEM), "Failed to open BPF object");
+
+        r = bpf_map__resize(obj->maps.ifaces_map, MAX(set_size(restrict_network_interfaces), 1u));
+        if (r != 0)
+                return log_unit_error_errno(u, r,
+                                "Failed to resize BPF map '%s': %m",
+                                bpf_map__name(obj->maps.ifaces_map));
+
+        obj->rodata->is_allow_list = (__u8) is_allow_list;
+
+        r = restrict_ifaces_bpf__load(obj);
+        if (r != 0)
+                return log_unit_error_errno(u, r, "Failed to load BPF object");
+
+        map_fd = bpf_map__fd(obj->maps.ifaces_map);
+
+        SET_FOREACH(iface, restrict_network_interfaces) {
+                ifindex = resolve_ifname(NULL, iface);
+                if (ifindex < 0) {
+                        log_unit_warning(u, "Couldn't find index of network interface, ignoring '%s'", iface);
```

Comment:
log_unit_warning_errno() please, and pass ifindex as error code. if it is negative it contains the error code. also, please include %m in message